internal/engine/upper_test.go

```diff
@@ -2801,9 +2803,22 @@ func TestWatchManifestsWithCommonAncestor(t *testing.T) {
 	call = f.nextCall("m1 build2")
 	assert.Equal(t, m1.K8sTarget(), call.k8s())
 
+	imageStateM1I0 := call.state[m1.ImageTargets[0].ID()]
```

Comment:
this test seems of a piece with the tests above in `buildcontroller_test.go`, any reason it's here instead of there? nbd, just seems odd.

Reply:
ya, probably. i think a lot of these need to be reorganized anyway